### PR TITLE
refactor: replace typeorm by mongoose

### PR DESCRIPTION
**Pull request by nunocaseiro**
Relates to #67

## Proposed Changes

  - Refactor the backend in order to use mongoose replacing typeorm.
  - Apply the new data structure as mentioned in the previous [discussion](https://github.com/orgs/xgeekshq/teams/d-c-backend/discussions/5)

### CODE REVIEW

File: backend/src/models/boards/dto/card/card.dto.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { IsNotEmpty, IsString, ValidateNested } from 'class-validator';
+import { Transform, TransformFnParams } from 'class-transformer';
+import CardItem from '../../schemas/card.item.schema';
+
+export default class CardDto {
+  @IsNotEmpty()
+  @IsString()
+  _id?: string;
```
Comment (rpvsilva) [inline backend/src/models/boards/dto/card/card.dto.ts:8]: You have `_id?: string` but also have `@IsNotEmpty()` annotation, that makes sense?

File: backend/src/models/boards/dto/card/card.dto.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  @IsNotEmpty()
+  @ValidateNested({ each: true })
+  items!: CardItem[];
+
+  // @ValidateNested({ each: true })
+  // @Type(() => CommentDto)
+  // comments!: CommentDto[];
+
+  // @IsNotEmpty()
+  // votes!: string[];
```
Comment (rpvsilva) [inline backend/src/models/boards/dto/card/card.dto.ts:24]: This is supposed to be commented?

File: backend/src/models/boards/dto/card/card.item.dto.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { IsNotEmpty, IsString } from 'class-validator';
+import { Transform, TransformFnParams } from 'class-transformer';
+
+export default class CardItemDto {
+  @IsNotEmpty()
+  @IsString()
+  _id?: string;
```
Comment (rpvsilva) [inline backend/src/models/boards/dto/card/card.item.dto.ts:7]: Same as above

File: backend/src/models/boards/dto/card/card.item.dto.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  @IsString()
+  @Transform(({ value }: TransformFnParams) => value.trim())
+  text!: string;
+
+  // @ValidateNested({ each: true })
+  // @Type(() => CommentDto)
+  // comments!: CommentDto[];
+
+  // @IsNotEmpty()
+  // votes!: string[];
```
Comment (rpvsilva) [inline backend/src/models/boards/dto/card/card.item.dto.ts:19]: Same as above

File: backend/src/models/boards/dto/board.dto.ts [hunk 5]
```diff
@@ -6,17 +6,15 @@ import {
...
 import { Optional } from '@nestjs/common';
-import { ObjectID } from 'typeorm';
 import { Transform, TransformFnParams, Type } from 'class-transformer';
-import ColumnDto from './column.dto';
-import UserDto from '../../users/dto/user.dto';
-import CardDto from './card.dto';
+import ColumnDto from './column/column.dto';
 
 export default class BoardDto {
   @Optional()
```
Comment (rpvsilva) [inline backend/src/models/boards/dto/board.dto.ts:16]: Here you are using the `Optional()` annotation from `@nestjs/common` and the other DTOs you are using from `class-validator` the `IsOptional()`. Does this makes sense?

File: backend/src/models/users/dto/user.dto.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { IsNotEmpty, IsOptional, IsString } from 'class-validator';
+
+export default class UserDto {
+  @IsOptional()
+  _id!: string;
```
Comment (rpvsilva) [inline backend/src/models/users/dto/user.dto.ts:5]: `@IsOptional()` and then you have `_id!: string`

File: backend/src/models/users/schemas/user.schema.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+export type UserDocument = User & Document;
+
+@Schema()
+export default class User {
+  @Prop({ nullable: false })
+  name!: string;
+
+  @Prop({ nullable: false })
+  password?: string;
```
Comment (rpvsilva) [inline backend/src/models/users/schemas/user.schema.ts:11]: Is this not nullable?

File: backend/src/models/users/users.service.ts [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    throw new HttpException(TOKENS_NOT_MATCHING, 401);
+  }
+
+  create(userData: CreateUserDto) {
+    return this.userModel.create(userData);
+  }
+
+  async delete(email: string) {
+    await this.userModel.deleteOne({ email }).exec();
+  }
```
Comment (rpvsilva) [inline backend/src/models/users/users.service.ts:51]: Is this a good approach? Deleting and not doing a return? I think you could do a `return;`

File: frontend/.gitignore [hunk 9]
```diff
@@ -4,6 +4,8 @@
 /node_modules
 /.pnp
 .pnp.js
+package-lock.json
+yarn.lock
```
Comment (rpvsilva) [inline frontend/.gitignore:8]: Same as backend

**Review by rpvsilva — CHANGES_REQUESTED** [review]

File: backend/src/models/boards/dto/card/card.dto.ts (hunk 2 above)
Comment (nunocaseiro) [reply backend/src/models/boards/dto/card/card.dto.ts:24]: Yes :) I am setting up the dto to introduce the comments and votes on the following pull requests

File: backend/src/models/boards/dto/board.dto.ts (hunk 5 above)
Comment (nunocaseiro) [reply backend/src/models/boards/dto/board.dto.ts:16]: yes, thanks!

**Review by rpvsilva — APPROVED** [review]
LGTM 🔥

**Pull request merged**